jaxrs-api/src/main/java/javax/ws/rs/core/EntityTag.java

```diff
@@ -99,17 +100,13 @@ public String getValue() {
      */
     @Override
     public boolean equals(final Object obj) {
-        if (obj == null) {
+        if (obj == null || !(obj instanceof EntityTag)) {
```

Comment:
`null instanceof <class>` is guaranteed to be `false` by definition, so you should omit the redundant check.